rclpy/rclpy/task.py

```diff
@@ -79,6 +87,13 @@ def result(self):
 
         :return: The result set by the task
         """
+        if self._cancelled:
+            raise CancelledError
+        # a coroutine might not have finished after calling spin_once
+        # if not self._done:
+        #     raise Error
```

Comment:
Is this the right behavior? I believe that in the other Future's Python provides, calling result before being set is an error (you should check if it is done first).

If calling `result` prematurely is an issue, and we only care about whether or not there was an exception, why not just do something like this in the executors:

```python
if future.exception() is not None:
    raise future.exception()
```

Reply:
> Is this the right behavior?

This is the current behavior of our coroutines / spinning them. The commented code is what Python's future class does but we can't enforce that at the moment.

> why not just do something like this in the executors

We can do that instead. We don't even have to change the `result()` method at all for now.

Reply:
I removed all the checking from `result()` for now. We should revisit this though after the release (see follow up ticket #200).